controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -348,17 +351,27 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases() (reconcile.Resul
 				}
 				storageClassClaimNamespacedName := r.getNamespacedName()
 
-				if annotations := storageClass.GetAnnotations(); annotations == nil {
-					storageClass.SetAnnotations(map[string]string{
-						StorageClassClaimAnnotation: storageClassClaimNamespacedName,
-					})
-				} else {
-					annotations[StorageClassClaimAnnotation] = storageClassClaimNamespacedName
-				}
+				addAnnotation(storageClass, StorageClassClaimAnnotation, storageClassClaimNamespacedName)
 				err = r.createOrReplaceStorageClass(storageClass)
 				if err != nil {
 					return reconcile.Result{}, fmt.Errorf("failed to create or update StorageClass: %s", err)
 				}
+
+			case "VolumeSnapshotClass":
+				var volumeSnapshotClass *snapapi.VolumeSnapshotClass
+				data["csi.storage.k8s.io/snapshotter-secret-namespace"] = r.storageClassClaim.Namespace
+
+				if resource.Name == "cephfs" {
+					volumeSnapshotClass = r.getCephFSVolumeSnapshotClass(data)
+				} else if resource.Name == "ceph-rbd" {
+					volumeSnapshotClass = r.getCephRBDVolumeSnapshotClass(data)
+				}
+				storageClassClaimNamespacedName := r.getNamespacedName()
+
+				addAnnotation(volumeSnapshotClass, StorageClassClaimAnnotation, storageClassClaimNamespacedName)
```

Comment:
Now that you have the new util func, we can just do: 
```suggestion
				addAnnotation(volumeSnapshotClass, StorageClassClaimAnnotation, r.getNamespacedName())
```

You can also apply the same for the storageclass case